addons/hr_timesheet/wizard/hr_employee_delete_wizard.py

```diff
@@ -0,0 +1,66 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import api, fields, models, _
+
+
+class HrEmployeDeleteWizard(models.TransientModel):
+    _name = 'hr.employee.delete.wizard'
+    _description = 'Employee Delete Wizard'
+
+    employee_ids = fields.Many2many('hr.employee', string='Employees')
+    has_active_employee = fields.Boolean(string='Has Active Employee', compute='_compute_has_active_employee')
+    has_timesheet = fields.Boolean(string='Has Timesheet', compute='_compute_has_timesheet', compute_sudo=True)
+
+    @api.depends('employee_ids')
+    def _compute_has_timesheet(self):
+        self = self.with_context(active_test=False)
+        timesheet_read_group = self.env['account.analytic.line']._read_group([
+            ('employee_id', 'in', self.employee_ids.ids)],
+            ['employee_id'], ['employee_id']
+        )
+        timesheet_employee_map = set(employee_info['employee_id'][0] for employee_info in timesheet_read_group)
+        for wizard in self:
+            wizard.has_timesheet = timesheet_employee_map & set(wizard.employee_ids.ids)
+
+    @api.depends('employee_ids')
+    def _compute_has_active_employee(self):
+        for wizard in self:
+            wizard.has_active_employee = wizard.with_context(active_test=True).employee_ids
+
+    def action_archive(self):
+        self.ensure_one()
+        if len(self.employee_ids) != 1:
+            return self.employee_ids.toggle_active()
+        return {
+            'name': _('Employee Termination'),
+            'type': 'ir.actions.act_window',
+            'res_model': 'hr.departure.wizard',
+            'views': [[False, 'form']],
+            'view_mode': 'form',
+            'target': 'new',
+            'context': {
+                'active_id': self.employee_ids.id,
+                'toggle_active': True,
+            }
+        }
```

Comment:
this condition is not already managed in the `toggle_active` method in `hr.employee`? 🤔 
See here:
https://github.com/odoo/odoo/blob/39c14368a2782365a2bf0649c26d9fe4749e67af/addons/hr/models/hr_employee.py#L414-L424

Reply:
If it is just the name and the context to change we could change the dict that we receive with the toggle_active method. What do you think? 🤷‍♂️🧐

Reply:
`toggle_active` of `hr` call super method, so even without clicking on `Apply` button of wizard it'll archive the record [video](https://tinyurl.com/292kq7dp)

Reply:
Oh okay I see, so if I correctly understood we want to have the same behavior than the one in JS you shared me. LGTM then, thanks for the explanation! 😃